backend/src/indexer/block_preprocessor.rs

```diff
@@ -106,6 +108,23 @@ impl NodeMetricLabels {
     }
 }
 
+/// Cache holding the last computed total stake for a block height
+#[derive(Clone)]
+struct TotalStakedAmountsCache {
+    block_height: u64,
+    total_staked: Amount,
+}
+
+/// A read write lock for the total staked amounts cache
+static TOTAL_STAKED_AMOUNTS_CACHE: Lazy<Arc<RwLock<Option<TotalStakedAmountsCache>>>> =
+    Lazy::new(|| Arc::new(RwLock::new(None)));
+
+/// How often to recompute the staked amounts for validators, the default
+/// interval to recompute staked amounts is every X blocks.
+pub static STAKE_RECOMPUTE_INTERVAL_IN_BLOCKS: Lazy<u64> = Lazy::new(|| {
+    env::var("STAKE_RECOMPUTE_INTERVAL").ok().and_then(|s| s.parse::<u64>().ok()).unwrap_or(500)
```

Comment:
If you want a new environmental variable, it should be part of the cli so it is displayed under the `--help` menu to the user as a configurable option. Hardcording `500` as default value here is impossible for users to discover/see or be aware of.

https://github.com/Concordium/concordium-scan/blob/main/backend/src/bin/ccdscan-indexer.rs#L28